src/util/stop.c

```diff
@@ -49,6 +75,8 @@ int nlopt_stop_f(const nlopt_stopping * s, double f, double oldf)
 int nlopt_stop_x(const nlopt_stopping * s, const double *x, const double *oldx)
 {
     unsigned i;
+    if (diff_norm(s->n, x, oldx, NULL) <= s->xtol_rel * diff_norm(s->n, x, oldx, NULL))
+        return 1;
     for (i = 0; i < s->n; ++i)
         if (!relstop(oldx[i], x[i], s->xtol_rel, s->xtol_abs[i]))
             return 0;
```

Comment:
I was thinking we would just pass `0` instead of `s->xtol_rel` here, i.e. only have the norm version of the `xtol_rel` test`.

Reply:
More simply, we could just test `if (fabs(x[i] - oldx[i]) > s->xtol_abs[i])`